### PR TITLE
Add tests for Avro OCF

### PR DESCRIPTION
**Pull request by ruchirK**

### CODE REVIEW

File: src/coord/coord.rs [hunk 1]
```diff
@@ -1423,6 +1422,21 @@ where
...
+                                SinkConnector::AvroOcf(AvroOcfSinkConnector { path }) => {
+                                    broadcast(
+                                        &mut self.broadcast_tx,
+                                        SequencedCommand::AppendLog(
+                                            MaterializedEvent::AvroOcfSink {
+                                                id: entry.id(),
+                                                path: path
+                                                    .to_str()
+                                                    .unwrap_or("<invalid utf8>")
+                                                    .to_string(),
```
Comment (benesch) [inline src/coord/coord.rs:1434]: I think `path` should be a `Vec<u8>`, so that we can handle this properly.

File: src/coord/coord.rs [hunk 2]
```diff
@@ -1653,6 +1667,16 @@ where
                     }),
                 );
             }
+            SinkConnector::AvroOcf(AvroOcfSinkConnector { path }) => {
+                broadcast(
+                    &mut self.broadcast_tx,
+                    SequencedCommand::AppendLog(MaterializedEvent::AvroOcfSink {
+                        id,
+                        path: path.to_str().unwrap_or("<invalid utf8>").to_string(),
```
Comment (benesch) [inline src/coord/coord.rs:1675]: Ditto here.

File: src/dataflow-types/logging.rs [hunk 3]
```diff
@@ -306,6 +311,11 @@ impl LogVariant {
                 .add_column("global_id", ScalarType::String)
                 .add_column("topic", ScalarType::String)
                 .add_keys(vec![0]),
+
+            LogVariant::Materialized(MaterializedLog::AvroOcfSinks) => RelationDesc::empty()
+                .add_column("global_id", ScalarType::String)
+                .add_column("path", ScalarType::String)
```
Comment (benesch) [inline src/dataflow-types/logging.rs:317]: Per above, `ScalarType::Bytes` I think.

File: src/dataflow/logging/materialized.rs [hunk 4]
```diff
@@ -144,6 +153,16 @@ pub fn construct<A: Allocate>(
                         let time_ms = time_ms as Timestamp;
 
                         match datum {
+                            MaterializedEvent::AvroOcfSink { id, path, insert } => {
+                                avro_ocf_sinks_session.give((
+                                    Row::pack(&[
+                                        Datum::String(&id.to_string()),
+                                        Datum::String(&path),
```
Comment (benesch) [inline src/dataflow/logging/materialized.rs:160]: And `Datum::Bytes` here.

File: src/coord/coord.rs (hunk 1 above)
Comment (ruchirK) [reply src/coord/coord.rs:1434]: I'm happy to make the changes to do that - but then the output from this log will not be user friendly at all e.g.

```
materialize=> select path from mz_avro_ocf_sinks;
                                                                                                             path
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 \x2f7661722f666f6c646572732f6a702f305f6e797870796a3476786738783463316b783176686c3830303030676e2f542f2e746d707441317a6d642f62617369632d73696e6b2d7531312d313538353936363130312d353830353134363331353038343638313330332e6f6366
(1 row)

materialize=>
```

not sure which is worse?

**Pull request merged**